test/conftest.py

```diff
@@ -75,6 +79,32 @@ def pytest_addoption(parser):
     )
 
 
+def pytest_configure(config):
+    """Load dotenv before tests are run to set PUDL_OUTPUT and PUDL_CACHE."""
+    load_dotenv()
+
+
+def pudl_env(request, tmpdir_factory):
+    """Set PUDL_OUTPUT environment variable."""
+    tmpdir = tmpdir_factory.mktemp("PUDL_OUTPUT")
+    os.environ["PUDL_OUTPUT"] = str(tmpdir)
+    os.environ["DAGSTER_HOME"] = str(tmpdir)
+
+    # In CI we want a hard-coded path for input caching purposes:
+    if os.environ.get("GITHUB_ACTIONS", False):
+        os.environ["PUDL_CACHE"] = str(Path(os.environ["HOME"]) / "pudl-work")
+    # If --tmp-data is set, create a disposable temporary datastore:
+    elif request.config.getoption("--tmp-data"):
+        os.environ["PUDL_CACHE"] = str(tmpdir)
```

Comment:
Does `pudl-work` need to be created on the Github Action?

Reply:
I think the `pudl-work` directory may have previously been created by `pudl_setup`, so I think it probably does need to be created somewhere in the new arrangement. As the comment says it's hard-coded so that if none of the DOIs have changed between CI runs, we can just use the previously cached raw inputs from the datastore.

You might want to look at how those DOI changes are detected in the caching step within the `tox-pytest` workflow. I'm not sure if it needs to be adapted to the new setup.

Reply:
It looks like previously `pudl-work` was just being created directly by the action, and instead of using `pudl_setup`, the paths are just configured in the `pudl_settings` fixture.